notebook/templates/login.html

```diff
@@ -85,6 +85,24 @@
       <p>
         Cookies are required for authenticated access to notebooks.
       </p>
+      {% if allow_password_change %}
+        <h3>{% trans %}Setup a Password{% endtrans %}</h3>
+        <p> You can also setup a password by entering your token and a new password
+        on the fields below:</p>
+        <form action="{{base_url}}login?next={{next}}" method="post" class="">
+                {{ xsrf_form_html() | safe }}
+          <div class="form-group">
+            <input type="password" name="password" id="password_input" class="form-control" placeholder="Token">
+          </div>
+          <div class="form-group">
+            <input type="password" name="new_password" id="new_password_input"
+            class="form-control" placeholder="New password" required>
```

Comment:
Should we get the user to type the new password twice, and check that it's the same?

Reply:
I don't think it is necessary, they can still issue `jupyter notebook password` to reset.